pkg/chaosdaemon/stress_server.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosdaemon
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/containerd/cgroups"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
+	pb "github.com/pingcap/chaos-mesh/pkg/chaosdaemon/pb"
+)
+
+var (
+	stressorLocker = new(sync.Mutex)
+	podStressors   = make(map[string]*exec.Cmd)
```

Comment:
Save pid list here is not safe. If `chaos-daemon` restarts, we can never stop the chaos action. Why not use `-t` argument for `stress-ng` and do nothing when `Recover`? (Though this is also little dangerous.)

Reply:
Yes. Seems to be better to wrapps all of arguments of `stress-ng` with our own type of argument as you said below. I eases usiblity and isolates the underlying stressing generator. However, the power of `stress-ng` may be limited by our own type. I think it's a tradeoff.  What do you think ?

Reply:
Yes. It's safer than now. We could convert user specified `duration` into `-t`.

Reply:
Maybe we could wrap arguments of `stress-ng` with our own arguments ?

Reply:
Howabout we save pid here and keep it in `Status`?  When the chaos daemon restarts we could always cancel it. I think it's more reliable.

Reply:
Great 👍

Reply:
Nice catch ! If we delete the container without delete the chaos, I think the stressing instance is still there. If we delete the pod whithout delete the chaos, I'm not sure about it. I'll check it. Did you mean we should eliminate the stressing instance once the container or pod is destroyed ? @cwen0

Reply:
If the stress action takes no effect on other pods when the container or pod was deleted, it is unnecessary to eliminate the stressing instance.

Reply:
@AngleNet As I have tested on minikube, when the pod is removed, the `stress-ng` process is also removed. But I am not clear about this mechanism.